core/src/commonMain/kotlin/com/ustadmobile/core/domain/validateusername/ValidateUsernameUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package com.ustadmobile.core.domain.validateusername
+
+
+class ValidateUsernameUseCase {
+
+    /**
+     * Validates a username according to the following rules:
+     * - Must not contain spaces, tabs, or banned special characters like !@#$%^&*()
+     * - Non-English Unicode characters are allowed
+     * - Must be 3–15 characters long after trimming
+     *
+     * @param username The username string to validate
+     * @return The valid username (trimmed) if valid, or null if invalid
+     */
+    operator fun invoke(username: String): String? {
+        val trimmed = username.trim()
+
+        if (trimmed.length !in 3..15) {
```

Comment:
Clear violation of coding style rule: "never hardcode literals" see https://github.com/UstadMobile/UstadMobile/blob/primary/CODING-STYLE.md#never-hardcode-any-literal-values